### PR TITLE
Breakpoint Bug Fixes

### LINKED ISSUE #575: Breakpoint's hit count always is 0 when debugging
Repro steps:
1. Add some breakpoints for code in Script.ps1 file.
2. F5
3. we can see the Breakpoint's information in Breakpoins window.

Expect: 
when hitting a breakpoint in Editor window, the current hit counts should not 0 in Breakpoint window

Actual: 
The current hit counts is still 0. you can have a look below picture.
hitcounts

### PR DESCRIPTION
**Pull request by MrayerMSFT**
Resolves #575 and adamdriscoll/poshtools#254

Summary of changes:
1. Breakpoints are now removed from the _psBreakpointTable when we remove breakpoints via debugging commands (see RemoveBreakpointById in PowerShellDebuggingService.cs)
2. Breakpoints are added to the breakpoint manager's _breakpoints table when they are set
3. _breakpoints is cleared when ClearBreakpoints is called
4. Call to GetTrueFileName is now in each breakpoint manager method
5. Added code for breakpoint counting
6. BreakpointRemoved is actually called now!

@HuanhuanSunMSFT @AndreSayreMSFT

### CODE REVIEW

File: PowerShellTools/DebugEngine/ScriptBreakpoint.cs [hunk 1]
```diff
@@ -42,6 +44,26 @@ public ScriptBreakpoint(ScriptProgramNode node, string file, int line, int colum
             Line = line;
             Column = column;
             File = file;
+            _hitCount = 0;
+        }
+
+        public override bool Equals(object obj)
```
Comment (ASayre) [inline PowerShellTools/DebugEngine/ScriptBreakpoint.cs:50]: When overriding Equals, we should also override GetHashCode

Comment (MrayerMSFT) [inline PowerShellTools/DebugEngine/ScriptBreakpoint.cs:50]: GetHashCode now overridden in 43fa56f.

**Comment by ASayre** [comment]
Intentional that _node is part of the hash code but not part of equality?

**Comment by MrayerMSFT** [comment]
Just wanted to give the hash code some more randomness. That's all. No good reason to not have _node in equality though, so I'll update that real quick.

**Comment by HuanhuanSunMSFT** [comment]
Good work.

**Pull request merged**